xrtl/examples/triangle_example.cc

```diff
@@ -0,0 +1,352 @@
+      return false;
+    }
+
+    // Acquire a framebuffer to render into.
+    auto framebuffer_ready_fence = context_->CreateQueueFence();
+    ref_ptr<ImageView> framebuffer_image_view;
+    auto acquire_result = swap_chain_->AcquireNextImage(
+        framebuffer_ready_fence, &framebuffer_image_view);
+    switch (acquire_result) {
+      case SwapChain::AcquireResult::kSuccess:
+        break;
+      case SwapChain::AcquireResult::kResizeRequired:
+        LOG(WARNING) << "Swap chain resize required";
+        break;
+      default:
+        LOG(ERROR) << "Failed to acquire framebuffer";
+        return false;
+    }
+
+    // TODO(benvanik): cache framebuffers for each image view.
+    auto framebuffer = context_->CreateFramebuffer(
+        render_pass_, framebuffer_image_view->size(), {framebuffer_image_view});
+    if (!framebuffer) {
+      LOG(ERROR) << "Unable to create framebuffer";
+      return false;
+    }
+
+    // Draw triangle.
+    auto rpe = command_buffer->BeginRenderPass(render_pass_, framebuffer);
+    rpe->SetViewport({0, 0, 640, 480});  // TODO(benvanik): query.
+    rpe->BindPipeline(render_pipeline_);
+    rpe->BindVertexBuffers(0, {triangle_buffer_});
+    rpe->Draw(3);
+    command_buffer->EndRenderPass(std::move(rpe));
+
+    // Submit command buffer for drawing the triangle. We wait until the
+    // framebuffer is ready for rendering.
+    auto render_complete_fence = context_->CreateQueueFence();
+    auto submit_result =
+        context_->Submit(std::move(framebuffer_ready_fence),
+                         std::move(command_buffer), render_complete_fence);
+    switch (submit_result) {
+      case Context::SubmitResult::kSuccess:
+        break;
+      default:
+        LOG(ERROR) << "Failed to submit rendering commands";
+        return false;
+    }
+
+    // Submit the framebuffer for presentation.
+    auto present_result = swap_chain_->PresentImage(
+        std::move(render_complete_fence), framebuffer_image_view);
+    switch (present_result) {
+      case SwapChain::PresentResult::kSuccess:
+        break;
+      case SwapChain::PresentResult::kResizeRequired:
+        LOG(WARNING) << "Swap chain resize required";
+        break;
+      default:
+        LOG(ERROR) << "Failed to present framebuffer";
+        return false;
+    }
+
+    return true;
+  }
+
+  void OnError(ref_ptr<Control> target) override {
+    LOG(INFO) << "OnError";
+    done_event_->Set();
+  }
+
+  void OnCreating(ref_ptr<Control> target) override {
+    LOG(INFO) << "OnCreating";
+  }
+
+  void OnCreated(ref_ptr<Control> target) override { LOG(INFO) << "OnCreated"; }
+
+  void OnDestroying(ref_ptr<Control> target) override {
+    LOG(INFO) << "OnDestroying";
+  }
+
+  void OnDestroyed(ref_ptr<Control> target) override {
+    LOG(INFO) << "OnDestroyed";
+    done_event_->Set();
+  }
+
+  void OnSystemThemeChanged(ref_ptr<Control> target) override {
+    LOG(INFO) << "OnSystemThemeChanged";
```

Comment:
probably not - I'm leaving them in until the demo is complete and things like resizing are done - pretty sure system theme changed won't be needed, but not sure about some of the others.